interface-impl-amf/src/test/java/org/mule/amf/impl/ApiParserAmfTestCase.java

```diff
@@ -6,45 +6,108 @@
+  private static final String APPLICATION_JSON = "application/json";
+  private static final String STATUS_200 = "200";
 
-  @Before
-  public void setup() throws Exception {
-    String pathAsUri =
-            requireNonNull(getClass().getClassLoader().getResource("org/mule/amf/impl/ref-json-schema/input.raml")).toString();
-    amfApiParser = new AMFParser(ApiReference.create(pathAsUri), true);
+  @Parameterized.Parameter
+  public ApiVendor apiVendor;
+
+  @Parameterized.Parameter(1)
+  public AMFParser amfApiParser;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection apiSpecifications() throws Exception {
+    String apiLocation = ApiParserAmfTestCase.class.getResource("references/api.raml").toURI().toString();
+    ApiReference ramlApiRef = ApiReference.create(apiLocation);
+
+    apiLocation = ApiParserAmfTestCase.class.getResource("references/oas20-api.yaml").toURI().toString();
+    ApiReference oas20apiRef = ApiReference.create(apiLocation);
+
+    apiLocation = ApiParserAmfTestCase.class.getResource("references/oas30-api.yaml").toURI().toString();
+    ApiReference oas30apiRef = ApiReference.create(apiLocation);
+
+    return Arrays.asList(new Object[][]{
+            {ApiVendor.RAML, new AMFParser(ramlApiRef, false)},
+            {ApiVendor.OAS_20, new AMFParser(oas20apiRef, false)},
+            {ApiVendor.OAS_30, new AMFParser(oas30apiRef, false)}
+    });
   }
 
   @Test
   public void testGetAllReferences() {
     List<String> references = amfApiParser.parse().getAllReferences();
 
-    assertThat(references.size(), is(2));
-    assertThat(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/ref-json-schema/car-schema.json")), is(true));
-    assertThat(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/ref-json-schema/car-properties-schema.json")),
-            is(true));
+    if (ApiVendor.RAML.equals(apiVendor)) {
+      assertEquals(9, references.size());
+    } else if (ApiVendor.OAS_20.equals(apiVendor)) {
+      assertEquals(2, references.size());
+    } else if (ApiVendor.RAML.equals(apiVendor)) {
+      assertEquals(3, references.size());
+    }
+    assertTrue(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/references/car-schemas/car-properties-schema.json")));
+    if (ApiVendor.RAML.equals(apiVendor)) {
+      assertTrue(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/references/person-schema.json")));
+      assertTrue(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/references/car-schemas/manufacturer-schema.json")));
+      assertTrue(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/references/car-schema.json")));
+      assertTrue(references.stream().anyMatch(ref -> ref.endsWith("org/mule/amf/impl/references/examples/user-example-for-raml.json")));
+      // TODO uncomment after APIMF-2102 and APIMF-2084 are fixed. AMF is not resolving references of references for OAS 20 and 30
```

Comment:
uncomment ?

Reply:
My bad, I fixed the comment to be clearer.